File: src/main/java/com/example/item/ModItemIds.java
package com.example.item;

import com.example.ExampleMod;

import net.minecraft.core.registries.Registries;
import net.minecraft.resources.Identifier;
import net.minecraft.resources.ResourceKey;
import net.minecraft.world.item.Item;

public class ModItemIds {
    public static final ResourceKey<Item> SUSPICIOUS_SUBSTANCE = create("suspicious_substance");
    public static final ResourceKey<Item> SWIFT_BRIDGE_ITEM = create("swift_bridge_item");
    public static final ResourceKey<Item> WALKING_SPEED_ITEM = create("walking_speed_item");
    public static final ResourceKey<Item> BLOCK_BREAKER_ITEM = create("block_breaker_item");
    public static final ResourceKey<Item> IOSDC_BADGE_ITEM = create("iosdc_badge_item");

    public static ResourceKey<Item> create(String name) {
        return ResourceKey.create(Registries.ITEM, Identifier.fromNamespaceAndPath(ExampleMod.MOD_ID, name));
    }
}

File: Sources/SwiftJavaModFabric/ModItem/ModItemIds.swift
import MinecraftJavaAPI
import SwiftJava

public enum ModItemIds {
    public static var suspiciousSubstance: ResourceKey<Item> {
        create("suspicious_substance")
    }

    public static var swiftBridgeItem: ResourceKey<Item> {
        create("swift_bridge_item")
    }

    public static var walkingSpeedItem: ResourceKey<Item> {
        create("walking_speed_item")
    }

    public static var blockBreakerItem: ResourceKey<Item> {
        create("block_breaker_item")
    }

    public static var iosdcBadgeItem: ResourceKey<Item> {
        create("iosdc_badge_item")
    }

    static var pureSwiftItem: ResourceKey<Item> {
        create("pure_swift_item")
    }

    private static func create(_ name: String) -> ResourceKey<Item> {
        let identifierClass = try! JavaClass<Identifier>()
        let registriesClass = try! JavaClass<Registries>()
        let resourceKeyClass = try! JavaClass<ResourceKey<Item>>()

        let identifier = identifierClass.fromNamespaceAndPath(modID, name)

        return resourceKeyClass.create(
            registriesClass.ITEM,
            identifier
        )
    }
}